packages/rocketchat-lib/lib/roomTypesCommon.js

```diff
@@ -81,5 +97,70 @@ this.roomTypesCommon = class {
 		}
 		return FlowRouter.go(this.roomTypes[roomType].route.name, routeData, queryParams);
 	}
+
+	_addDefaultImplementations(config) {
+		if (!config.getDisplayName) {
+			config.getDisplayName = function(room) {
+				return room.name;
+			};
+		}
+
+		if (!config.allowChangeChannelSettings) {
+			config.allowChangeChannelSettings = function() {
+				return true;
+			};
+		}
+
+		if (!config.canBeDeleted) {
+			config.canBeDeleted = function(room) {
+				return Meteor.isServer ?
```

Comment:
@graywolf336 Unfortunately, the client- and server-side-method-signatures vary.
I preferred having this switch in the generic code compared to having dedicated client-side-files or redundantly implementing this. WDYT?

Reply:
:+1: